lib/charms/opensearch/v0/opensearch_locking.py

```diff
@@ -205,8 +205,8 @@ def _unit_with_lock(self, host) -> str | None:
                 retries=3,
             )
         except OpenSearchHttpError as e:
-            if e.response_code == 404:
-                # No unit has lock
+            if e.response_code in [404, 503]:
+                # No unit has lock or index not available
```

Comment:
As we've discussed, I think this is actually masking the problem. 503 means shards have not been alllocated for whatever reason.

Reply:
I agree. The idea of this change here is just to avoid raising an error to avoid deadlocks when removing the application.